src/extras.jl

```diff
@@ -142,36 +152,70 @@ function cut(x::AbstractArray{T, N}, breaks::AbstractVector;
 
     n = length(breaks)
     if labels isa Function
-        @static if VERSION >= v"0.7.0-DEV.4524"
-            from = map(x -> sprint(show, x, context=:compact=>true), breaks[1:n-1])
-            to = map(x -> sprint(show, x, context=:compact=>true), breaks[2:n])
-        else
-            from = map(x -> sprint(showcompact, x), breaks[1:n-1])
-            to = map(x -> sprint(showcompact, x), breaks[2:n])
-        end
+        from = map(x -> sprint(show, x, context=:compact=>true), breaks[1:n-1])
+        to = map(x -> sprint(show, x, context=:compact=>true), breaks[2:n])
         levs = Vector{String}(undef, n-1)
         for i in 1:n-2
-            levs[i] = labels(from[i], to[i], i, closed=false)
+            levs[i] = labels(from[i], to[i], i,
+                             leftclosed=breaks[i] != breaks[i+1], rightclosed=false)
         end
-        levs[end] = labels(from[end], to[end], n-1, closed=extend)
+        levs[end] = labels(from[end], to[end], n-1,
+                           leftclosed=breaks[end-1] != breaks[end], rightclosed=extend)
     else
         length(labels) == n-1 || throw(ArgumentError("labels must be of length $(n-1), but got length $(length(labels))"))
         # Levels must have element type String for type stability of the result
         levs::Vector{String} = copy(labels)
     end
+    if !allunique(levs)
+        if labels === default_formatter
+            throw(ArgumentError("all labels must be unique, but `breaks` contains duplicates: " *
```

Comment:
this might also happen if `sprint` produces duplicates for different values. maybe in https://github.com/JuliaData/CategoricalArrays.jl/pull/245/files#diff-0e27ad0a2d2d952a6b4e82aa6cf9a5daR160 and https://github.com/JuliaData/CategoricalArrays.jl/pull/245/files#diff-0e27ad0a2d2d952a6b4e82aa6cf9a5daR163 we should compare `from` and `to` instead.

See e.g.:
```
julia> sprint(show, 1.000001, context=:compact=>true)
"1.0"

julia> sprint(show, 1.000002, context=:compact=>true)
"1.0"

julia> sprint(show, 1.000003, context=:compact=>true)
"1.0"
```

Reply:
Good catch. But even in that case we should not allow duplicate labels. The solution would be to generate new labels with a higher precision. I'd say we can leave this for later as it won't break anything and it's mostly orthogonal to this PR.